ckanext/spectrum/actions.py

```diff
@@ -44,36 +45,60 @@ def package_create(next_action, context, data_dict):
 
     return next_action(context, data_dict)
 
+
 @toolkit.chained_action
 def user_create(next_action, context, data_dict):
-    """
-    Autogenerates a password and username (if password or username is not provided).
-    """
-
     if not data_dict.get('password'):
         data_dict['password'] = secrets.token_urlsafe(32)
 
-    if data_dict.get('name'):
-        return next_action(context, data_dict)
-
-    if not data_dict.get('email'):
-        raise toolkit.ValidationError(toolkit._("Must specify either a name or an email"))
+    if not data_dict.get('name'):
+        if not data_dict.get('email'):
+            raise toolkit.ValidationError(toolkit._("Must specify either a name or an email"))
+        else:
+            email = data_dict['email']
+            username = _get_random_username_from_email(email, context['model'])
+            data_dict['name'] = username
 
-    email = data_dict['email']
-    username = _get_random_username_from_email(email, context['model'])
-    data_dict['name'] = username
+    check_id_is_unique(context, data_dict)
 
     created_user = next_action(context, data_dict)
 
+    assign_user_to_default_organisation(context, created_user)
+
+    return created_user
+
+
+def assign_user_to_default_organisation(context, created_user):
     default_org_name = toolkit.config.get('ckanext.spectrum.default_organization', 'spectrum')
     org_member_dict = {'id': default_org_name, 'username': created_user['name'], 'role': 'editor'}
+
     try:
         ignore_auth_context = {"user": context["user"], "ignore_auth": True}
         toolkit.get_action('organization_member_create')(ignore_auth_context, org_member_dict)
     except toolkit.ValidationError:
         log.error(f"Failed to add newly created user: {created_user['name']} to org: {default_org_name}. "
                   f"User account got created successfully.")
-    return created_user
+
```

Comment:
Typo?  Looks like you have deleted a return statement without reason here?

Reply:
'created_user' is just a parameter passed to some functions here, it's not modified, so return is unnecessary.

Reply:
Ah... I missed your refactoring.  in fact created_user is returned now on line 68.  Sure that makes sense now.